core/internal/runhistoryreader/parquet/parition_reader.go

```diff
@@ -0,0 +1,184 @@
+package parquet
+
+// This file copies most of its logic from how history parquet files
+// are read on the backend.
+// See: https://github.com/wandb/core/blob/master/services/gorilla/internal/parquet/partition_reader.go
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/apache/arrow-go/v18/parquet"
+	"github.com/apache/arrow-go/v18/parquet/metadata"
+	"github.com/apache/arrow-go/v18/parquet/pqarrow"
+	"github.com/wandb/parallel"
+)
+
+// ParquetPartitionReader is a reader for a single run history parquet file.
+// Run history may be partitioned into multiple parquet files.
+type ParquetPartitionReader struct {
+	ctx       context.Context
+	ctxDone   <-chan struct{}
+	executor  parallel.Executor
+	file      *pqarrow.FileReader
+	dataReady chan struct{}
+	it        RowIterator
+	err       error
+}
+
+// Metadata returns the metadata of the underlying parquet partition file.
+func (p *ParquetPartitionReader) MetaData() (*metadata.FileMetaData, error) {
+	return p.file.ParquetReader().MetaData(), nil
+}
+
+// Next advances the reader to the next row in the row iterator.
+func (p *ParquetPartitionReader) Next() (bool, error) {
+	it, err := p.iterator()
+	if err != nil {
+		return false, err
+	}
+	p.it = it
+	return p.it.Next()
+}
+
+// Value returns the value of the current row in the row iterator.
+func (p *ParquetPartitionReader) Value() KVMapList {
+	return p.it.Value()
+}
+
+// Release releases the resources used by the reader.
```

Comment:
What are the expectations here? Is it required to call this when the reader is no longer used? Can it be called multiple times, or exactly once?